### ISSUE TITLE
Gaps in binding group indices

### ISSUE THREAD

**Issue by FredrikNoren** [gfx-rs/wgpu#1489]
**Description**
If I specify binding groups in wgsl (using reflection, no layout), I can specify for instance group 0 and 2 (and skip 1), but when I try to bind that in the application I get an error.

**Repro steps**
https://github.com/FredrikNoren/wgpu/tree/gap-bug/wgpu/examples/hello-triangle

**Expected vs observed behavior**
I'm getting the following error in the example: `current render pipeline has a layout which is incompatible with a currently set bind group, first differing at entry index 0`.

**Platform**
OSX

(@kvark requested that I filed this issue)

**Comment by jinleili** [comment]
In the [WebGPU standard](https://github.com/gpuweb/gpuweb/issues/2043), it is marked as `Milestone 2`.

**Comment by reima** [comment]
The current [WebGPU Candidate Recommendation Draft, 11 July 2025](https://www.w3.org/TR/2025/CRD-webgpu-20250711/) supports gaps in bind groups. Does this affect the priorization of this issue regarding WebGPU for Firefox?

**Comment by ErichDonGubler** [comment]
@reima: TBH, I'm not sure why we on the Firefox team marked this as `Out of scope` (@teoxoy, @jimblandy, do you remember?), but it's likely that this is still lower priority than many of the things we'll be working on for a while, unless there are applications already in the wild that are blocked on this.

**Comment by teoxoy** [comment]
We probably marked it as out of scope since at the time it wasn't part of the spec.

**Comment by jimblandy** [comment]
Are we sure this hasn't been fixed already? It seems like a big compat item.

**Comment by jimblandy** [comment]
> The current [WebGPU Candidate Recommendation Draft, 11 July 2025](https://www.w3.org/TR/2025/CRD-webgpu-20250711/) supports gaps in bind groups. Does this affect the priorization of this issue regarding WebGPU for Firefox?

Yes, for Mozilla, at least, it certainly is a higher priority to implement the spec than to implement extensions. Obviously contributors outside Mozilla add extensions all the time.

**Comment by reima** [comment]
> Are we sure this hasn't been fixed already? It seems like a big compat item.

I've adapted the repro of @FredrikNoren to the current trunk: https://github.com/reima/wgpu/tree/issue-1489/examples/features/src/hello_triangle

It errors out with a validation error:

```
thread 'main' panicked at wgpu\src\backend\wgpu_core.rs:2397:18:
wgpu error: Validation Error

Caused by:
  In a CommandEncoder
    In a draw command, kind: Draw
      The current set RenderPipeline with '' label expects a BindGroup to be set at index 0
```

So no, I don't think this has been fixed.
